frame/contracts-registry/src/tests.rs

```diff
@@ -0,0 +1,405 @@
+			vec![signer].try_into().unwrap()
+		));
+		assert_eq!(Balances::free_balance(2), 95u64);
+
+		let origin = RawOrigin::Signed(signer);
+		assert_ok!(ContractsRegistry::party_sign_contract(origin.into(), 0));
+		System::assert_last_event(Event::<Test>::PartySigned { contract_id: 0, signer }.into());
+	});
+}
+
+#[test]
+fn remove_judge_deposits_event() {
+	new_test_ext().execute_with(|| {
+		let judge = account("a", 0, SEED);
+		let origin = RuntimeOrigin::root();
+
+		for s in 0..<Test as Config>::MaxJudges::get() {
+			let user = account("user", s, SEED);
+			assert_ok!(ContractsRegistry::add_judge(origin.clone(), user));
+		}
+
+		assert_ok!(ContractsRegistry::remove_judge(origin.into(), judge));
+		System::assert_last_event(Event::<Test>::RemovedJudge { judge }.into());
+	});
+}
+
+#[test]
+fn creator_can_remove_contract() {
+	new_test_ext().execute_with(|| {
+		let origin = RuntimeOrigin::signed(2);
+
+		assert_eq!(Balances::free_balance(2), 100u64);
+		assert_ok!(ContractsRegistry::create_contract(
+			origin.clone(),
+			vec![].try_into().unwrap(),
+			vec![].try_into().unwrap()
+		));
+		assert_eq!(Balances::free_balance(2), 98u64);
+		assert!(Contracts::<Test>::get(0).is_some());
+		assert_ok!(ContractsRegistry::remove_contract(origin.into(), 0));
+		assert!(Contracts::<Test>::get(0).is_none());
+		assert_eq!(Balances::free_balance(2), 100u64);
+	});
+}
+
+#[test]
+fn remove_contract_deposits_event() {
+	new_test_ext().execute_with(|| {
+		let origin = RuntimeOrigin::signed(2);
+
+		assert_eq!(Balances::free_balance(2), 100u64);
+		assert_ok!(ContractsRegistry::create_contract(
+			origin.clone(),
+			vec![].try_into().unwrap(),
+			vec![].try_into().unwrap()
+		));
+		assert_eq!(Balances::free_balance(2), 98u64);
+		assert!(Contracts::<Test>::get(0).is_some());
+		assert_ok!(ContractsRegistry::remove_contract(origin.into(), 0));
+		assert!(Contracts::<Test>::get(0).is_none());
+		assert_eq!(Balances::free_balance(2), 100u64);
+		System::assert_last_event(Event::<Test>::ContractRemoved { contract_id: 0 }.into());
+	});
+}
+
+#[test]
+fn create_contract_successfully_reserve_deposit() {
+	new_test_ext().execute_with(|| {
+		let origin = RuntimeOrigin::signed(2);
+
+		assert_eq!(Balances::free_balance(2), 100u64);
+		assert_ok!(ContractsRegistry::create_contract(
+			origin.clone(),
+			vec![1, 2, 3].try_into().unwrap(),
+			vec![].try_into().unwrap()
+		));
+		assert_eq!(Balances::free_balance(2), 95u64);
+	});
+}
+
+#[test]
+fn remove_contract_successfully_unreserve_deposit() {
+	new_test_ext().execute_with(|| {
+		let origin = RuntimeOrigin::signed(2);
+
+		assert_eq!(Balances::free_balance(2), 100u64);
+		assert_ok!(ContractsRegistry::create_contract(
+			origin.clone(),
+			vec![1, 2, 3].try_into().unwrap(),
+			vec![].try_into().unwrap()
+		));
+		assert_eq!(Balances::free_balance(2), 95u64);
+		assert_ok!(ContractsRegistry::remove_contract(origin.clone(), 0));
+		assert_eq!(Balances::free_balance(2), 100u64);
+	});
+}
```

Comment:
Missing tests for checking if contracts can only be removed if there are no signatures at all - which I'm pretty sure will find a bug btw.